src/acceptance/broker/broker_test.go

```diff
@@ -164,80 +167,84 @@ var _ = Describe("AutoScaler Service Broker", func() {
-}
-
-type plans []string
+type ServicePlans []ServicePlan
 
-func (p plans) length() int { return len(p) }
-func (p plans) contains(planName string) bool {
-	for _, plan := range p {
-		if plan == planName {
-			return true
-		}
+type (
+	ServicePlan struct {
+		Guid          string        `json:"guid"`
+		Name          string        `json:"name"`
+		BrokerCatalog BrokerCatalog `json:"broker_catalog"`
 	}
-	return false
-}
+	BrokerCatalog struct {
+		Id       string   `json:"id"`
+		Features Features `json:"features"`
+	}
+	Features struct {
+		PlanUpdateable bool `json:"plan_updateable"`
+	}
+)
+
+func (p ServicePlans) length() int { return len(p) }
 
-func getPlans() plans {
-	values := url2.Values{
-		"fields[service_offering.service_broker]": []string{"name"},
-		"include":                []string{"service_offering"},
+func GetServicePlans(cfg *config.Config) ServicePlans {
+	values := url.Values{
 		"per_page":               []string{"5000"},
 		"service_broker_names":   []string{cfg.ServiceBroker},
 		"service_offering_names": []string{cfg.ServiceName},
 	}
-	url := &url2.URL{Path: "/v3/service_plans", RawQuery: values.Encode()}
-	getPlans := cf.CfSilent("curl", url.String(), "-f").Wait(cfg.DefaultTimeoutDuration())
-	Expect(getPlans).To(Exit(0), "failed getting plans")
+	servicePlansURL := &url.URL{Path: "/v3/service_plans", RawQuery: values.Encode()}
+
+	var result ServicePlans
+
+	// This should also work as normal user - for some reason, if we use the normal user
+	// plan_updateable is returned as integer instead of boolean
+	workflowhelpers.AsUser(setup.AdminUserContext(), cfg.DefaultTimeoutDuration(), func() {
+		serviceCmd := cf.CfSilent("curl", "-f", servicePlansURL.String()).Wait(cfg.DefaultTimeoutDuration())
+		Expect(serviceCmd).To(Exit(0), "failed getting service plans")
 
-	plansResult := &struct{ Resources []struct{ Name string } }{}
-	err := json.Unmarshal(getPlans.Out.Contents(), plansResult)
-	Expect(err).NotTo(HaveOccurred())
+		plansResult := &struct{ Resources []ServicePlan }{}
+		err := json.Unmarshal(serviceCmd.Out.Contents(), plansResult)
+		Expect(err).NotTo(HaveOccurred())
 
-	var p plans
-	for _, item := range plansResult.Resources {
-		p = append(p, item.Name)
+		result = plansResult.Resources
+	})
+
+	return result
+}
+
+func (p ServicePlan) isUpdatable() bool {
+	return p.BrokerCatalog.Features.PlanUpdateable
+}
+
+func (p ServicePlans) getSourceAndTargetForPlanUpdate() (source, target ServicePlan, err error) {
+	if p.length() < 2 {
+		return ServicePlan{}, ServicePlan{}, fmt.Errorf("two service plans needed, only one plan available")
+	}
+	updatablePlanIndex := slices.IndexFunc(p, func(n ServicePlan) bool {
+		return n.isUpdatable()
+	})
+	if updatablePlanIndex == -1 {
+		return ServicePlan{}, ServicePlan{}, fmt.Errorf("no updatable plan found")
 	}
-	return p
+	source = p[updatablePlanIndex]
+	target = p[(updatablePlanIndex+1)%p.length()]
```

Comment:
good old modulo 🤓👍  maybe a comment here helps to understand the intention 
```suggestion
	target = p[(updatablePlanIndex+1)%p.length()] // use any other plan than to update to
```